ofctrl/multipart_test.go

```diff
@@ -0,0 +1,76 @@
+// #nosec G404: random number generator not used for security purposes
+package ofctrl
+
+import (
+	"fmt"
+	"sync"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+
+	"antrea.io/libOpenflow/openflow13"
+)
+
+type multipartActor struct {
+	*OfActor
+	expectedTxs map[uint32]chan *openflow13.MultipartReply
+}
+
+func (o *multipartActor) MultipartReply(sw *OFSwitch, rep *openflow13.MultipartReply) {
+	if ch, found := o.expectedTxs[rep.Xid]; found {
+		ch <- rep
+	}
+}
+func TestMultipartReplyOrder(t *testing.T) {
+	app := new(multipartActor)
+	app.OfActor = new(OfActor)
+	app.expectedTxs = make(map[uint32]chan *openflow13.MultipartReply)
+	ctrl := NewController(app)
+	brName := "brMultipart"
+	ovsBr := prepareControllerAndSwitch(t, app.OfActor, ctrl, brName)
+	defer func() {
+		assert.Nilf(t, ovsBr.DeleteBridge(brName), "Failed to delete br %s", brName)
+		ctrl.Delete()
+	}()
+	testTableFeatures(t, app)
+}
+
+func testTableFeatures(t *testing.T, app *multipartActor) {
+	app.Switch.EnableMonitor()
+	ofSwitch := app.Switch
+	mpartRequest := &openflow13.MultipartRequest{
+		Header: openflow13.NewOfp13Header(),
+		Type:   openflow13.MultipartType_TableFeatures,
+		Flags:  0,
+	}
+	mpartRequest.Header.Type = openflow13.Type_MultiPartRequest
+	mpartRequest.Header.Length = mpartRequest.Len()
+	ch := make(chan *openflow13.MultipartReply)
+	xid := mpartRequest.Xid
+	app.expectedTxs[xid] = ch
+	ofSwitch.Send(mpartRequest)
+
+	wg := sync.WaitGroup{}
+	wg.Add(1)
+	tableCount := 0
+	go func(tableCount *int) {
+		defer wg.Done()
+		for {
+			select {
+			case rep := <-ch:
+				flags := rep.Flags
+				tableLen := len(rep.Body)
+				fmt.Printf("Receive Multipart reply with Flag %d, tables: %d\n", flags, tableLen)
+				*tableCount += tableLen
+				if flags == 0 {
+					return
+				}
+			}
+		}
+
+	}(&tableCount)
+	wg.Wait()
```

Comment:
nit: I am not sure why you need a separate goroutine and a wait group for this
you also do not need to pass `tableCount` as a parameter as it will be captured by the closure anyway

This should achieve the same thing with less complexity:

```go
getTableCount := func() {
    tableCount := 0
    for rep := range ch {
        flags := rep.Flags
        tableLen := len(rep.Body)
        t.Logf("Receive Multipart reply with Flag %d, tables: %d\n", flags, tableLen)
        tableCount += tableLen
        if flags == 0 {
            break
        }
    }
    return tableCount
}
tableCount := getTableCount()
```

Reply:
Updated.